src/main.rs

```diff
@@ -1,198 +1,116 @@
+use std::thread;
 use std::time::Duration;
-use std::{fs, thread};
 
-use gtfs_structures::Gtfs;
 use serde::{Deserialize, Serialize};
-use transit_board::config::Conf;
-use transit_board::mercury::MercuryDelays;
-use transit_board::{BusStopHandler, BusStopJson, Disruption, SubwayStopHandler, SubwayStopJson};
+use transit_board::config::{Conf, ServiceAlertsConf};
+use transit_board::{BusStopHandler, Disruption, Stop, SubwayStopHandler};
 use tungstenite::Message;
 
 fn main() {
-    dotenvy::dotenv().unwrap();
-    //let api_key = std::env::var("NYCTKEY").unwrap();
-    let api_key_bus = std::env::var("MTABUSKEY").unwrap();
-    let config = match fs::read_to_string("config.json") {
-        Ok(a) => a,
-        Err(_) => "none".to_owned(),
-    };
-    let config: Conf = match serde_json::from_str(&config) {
-        Ok(a) => a,
-        Err(_) => Default::default(),
-    };
-    let mut subway = config.get_subway_handlers();
-    let mut bus = config.get_bus_handlers(api_key_bus.clone());
-
-    let mut subway_map: HashMap<String, SubwayStopJson> = HashMap::new();
-    let mut bus_map: HashMap<String, BusStopJson> = HashMap::new();
-    let mut service_alerts_map: Vec<Disruption> = vec![];
-
-    let jerome = SubwayStopHandler::new(vec!["405S".to_string()], 10, true);
-    let concourse = SubwayStopHandler::new(vec!["D03S".to_string()], 14, true);
-    let bx1028 = BusStopHandler::new(
-        api_key_bus.to_owned(),
-        vec!["100017".to_string(), "103400".to_string()],
-        3,
-        false,
-    );
-    let bx222526 =
-        BusStopHandler::new(api_key_bus.to_owned(), vec!["100723".to_string()], 3, false);
-    if subway.len() == 0 {
-        subway = vec![jerome, concourse];
-    }
-    if bus.len() == 0 {
-        bus = vec![bx222526, bx1028];
-    }
+    let server = TcpListener::bind("0.0.0.0:9001").unwrap();
 
-    let send_data_clone = Arc::new(RwLock::new(InfoJson::default()));
-    let send_data = send_data_clone.clone();
-    let update_threads = Arc::new(RwLock::new(Vec::<JoinHandle<_>>::new()));
-    let update = update_threads.clone();
-
-    let gtfs = match Gtfs::from_url(
-        "http://web.mta.info/developers/data/nyct/subway/google_transit.zip",
-    ) {
-        Ok(a) => a,
-        Err(_) => return,
-    };
-
-    let update_thread = thread::spawn(move || {
-        loop {
-            subway_map.clear();
-            bus_map.clear();
-            service_alerts_map.clear();
-
-            for i in 0..subway.len() {
-                subway.get_mut(i).unwrap().refresh();
-                subway_map.insert(
-                    subway
-                        .get(i)
-                        .unwrap()
-                        .stop_ids
-                        .iter()
-                        .next()
-                        .unwrap()
-                        .to_owned(),
-                    subway.get(i).unwrap().serialize(),
-                );
-            }
+    for stream in server.incoming() {
```

Comment:
Ideally, I think a separate update thread makes sense, because then websockets would not get data that another one already got. That could be a simple hashmap where the websockets could add  their stops if they don't exist. Then in the loop they wait for an updater thread to finish and then they send data. I could implement that.

However, to be pragmatic, I don't think that we need to do that. Just because our use case has one server and one client in an essentially closed system. It might be worth doing later, but for now this works.